transactions/transactions.py

```diff
@@ -430,10 +530,46 @@ async def _send_member_message(self, ctx, member, message):
         command_prefix = ctx.prefix
         message = message.replace('[p]', command_prefix)
         message = message_title + message
-        try:
-            await member.send(message)
-        except:
-            await ctx.send(":x: Couldn't send message to this member.")
+        
+        await self.dm_helper_cog.add_to_dm_queue(member, content=message, ctx=ctx)
+
+    async def send_player_expire_contract_message(self, ctx: commands.Context, player: discord.Member, franchise_role: discord.Role, team: str, gm: discord.Member):
+        franchise_name = franchise_name = self.team_manager_cog.get_franchise_name_from_role(franchise_role)
+        msg = stringTemplates.contract_expiration_msg.format(
+            p=ctx.prefix, player=player, team=team, franchise=franchise_name, gm=gm.display_name
+        )
+        
+        embed = discord.Embed(title=f"Notice from {ctx.guild.name}", description=msg, color=discord.Color.blue())
+        if ctx.guild.icon_url:
+            embed.set_thumbnail(url=ctx.guild.icon_url)
+
+        await self.dm_helper_cog.add_to_dm_queue(member=player, embed=embed, ctx=ctx)
+
+    def _get_name_components(self, member: discord.Member):
+        if member.nick:
+            name = member.nick
+        else:
+            return "", member.name, ""
+        prefix = name[0:name.index(' | ')] if ' | ' in name else ''
+        if prefix:
+            name = name[name.index(' | ')+3:]
+        player_name = ""
+        awards = ""
+        for char in name[::-1]:
```

Comment:
a little hard to follow what's going on in this section, could you add a brief comment showing an example or explaining? I think that'd help a lot.

Reply:
Sure thing! The purpose of this function is to extract the `franchise_prefix`, `player_name` and `awards`. 

So league awards is defined somewhere as the stars, trophies, (medals from 1s). So `LEAGUE_AWARDS = [⭐, 🏆, 🥇]`. The purpose of this function is to extract the `franchise_prefix`, `player_name` and `awards`. 

It starts by splitting by the pipe character `|`, which yields the prefix.  `<0> | Nehtaro ⭐` is split into `<0>` and `Nehtaro ⭐`. Awards goes at the end of the name, so it starts by traversing the characters in reverse order and getting all rewards the player may have on their name. `name[::-1]` is what reverses the order of the name string. So `Nehtaro ⭐` becomes `⭐ oratheN`. So it tracks characters until it finds one that isn't defined in `LEAGUE_AWARDS`.

At the end of this, it will return `prefix.strip(), player_name.strip(), awards.strip()`. The strip just removes any excess spaces.

That gives you `<0>`, `Nehtaro`, `⭐`